### ISSUE TITLE
Problems setting inbound IMAP email

### ISSUE THREAD

**Issue by danielzzz** [chatwoot/chatwoot#3869]
**Describe the bug**

Hi, I have problems setting email inbox. I am not sure if I am missing something.

my idea is to connect chatwoot to an external email account with imap configuration.

I have create an inbox and setup the imap config, but no messages are being picked up.

do I need to configure something else?

I see the imap job is being scheduled then executed:
```
sidekiq_1   | I, [2022-01-28T19:00:17.716908 #1]  INFO -- : [ActiveJob] [Inboxes::FetchImapEmailsJob] [9b622c3b-80ec-4fd3-82a4-118b9093a0d3] Performing Inboxes::FetchImapEmailsJob (Job ID: 9b622c3b-80ec-4fd3-82a4-118b9093a0d3) from Sidekiq(low) enqueued at 2022-01-28T19:00:17Z with arguments: #<GlobalID:0x00007f4172b534d8 @uri=#<URI::GID gid://chatwoot/Channel::Email/5>>
sidekiq_1   | I, [2022-01-28T19:00:17.728161 #1]  INFO -- : [ActiveJob] [Account::ConversationsResolutionSchedulerJob] [b402763e-2a2d-47ec-8449-b00d951194be] Performed Account::ConversationsResolutionSchedulerJob (Job ID: b402763e-2a2d-47ec-8449-b00d951194be) from Sidekiq(scheduled_jobs) in 176.01ms
sidekiq_1   | 2022-01-28T19:00:17.728Z pid=1 tid=2bdp class=Account::ConversationsResolutionSchedulerJob jid=83332a8114bc079c3a5956cc elapsed=0.178 INFO: done
```
but nothing is being downloaded from the imap account.

I verified I can connect to the imap account with the settings provided.

do I need to configure something else? 

I am using the latest docker version on ubuntu 20lts.

thanks, dan

**Comment by danielzzz** [comment]
suddenly I started getting the emails, sorry for the noise..

**Comment by pranavrajs** [comment]
Glad to know it is solved, do you know anything which might have triggered this? This could be something we are missing on the docs as well.

**Comment by danielzzz** [comment]
hi @pranavrajs , thanks for reaching out.

I am not sure. Maybe it takes some time to kick in...? 
The first few emails I sent to the account were not retrieved / passed through - then suddenly it started working.
I will keep monitoring it, as it makes me a bit anxious some emails may get lost.

as for the setup / docs, these things were a bit confusing for me:
- what was the forwarding email for (how it worked)
- was the imap working or not (no feedback)
- was imap and smtp were both necessary (according to my tests, having just the imap set up allow for sending and receiving messages, right?). this part was quite confusing for me along with the sample .env file.

I was not sure if I needed some external service to receive / send mails from chatwoot (beside having another email account / server set up), as in the .env file there were a few options, but I was not quite clear to me how it worked all together.

It was not clear to me if chatwoot itself (or the docker-compose set up) self contained an email server that would receive the emails forwarder from another account or I definitely needed some other service (a local exim - not sure how to configure it with chatwoot, or external services). how this related to imap/smtp config.

What I think might help:
1) hide options that are mutually exclusive, for example:
if I just need imap to have everything working, I don't need to see other options.

2) for the forwarding option provide some short explanation text like:
"for this to work you need to have xxx set up, please refer to this link xxx"

3) if the smtp serves just to send messages out and needs to be combined with another option (that forwarding email?), please state so on the config page and maybe put them both together.

thought I guess having just smtp is useful when the user contacted you through another channel and you just want to send them notification through email if they provided one.

but for a first timer, it was not quite clear to me.

4) also VERY IMPORTANT (I think), please provide some way to test the config from the inbox config page.
for example if I setup imap, there should be an option to trigger mail fetching at the moment.
luckily I could check the logs, but still I could see it triggered and did not know if it could connect correctly, haven't seen any new messages or what happened

for smtp - this could mean an option to send out some test email. etc..

this could apply probably to other kinds of inboxes.

disclaimers:
* I am not a native speaker, maybe I did not understand the docs well enough
* I wish I could help more, but my ruby knowledge is very basic

Having said that, my first impressions with chatwoot are very good, I think it's a great piece of software. thank you very much for that!

**Comment by danielzzz** [comment]
generally these inboxes that are passive and managed by scheduled jobs, it would be great to have an option to trigger them manually, thanks!

**Comment by pranavrajs** [comment]
Reopening this issue as it seems that the product does not explain itself.

**Comment by pranavrajs** [comment]
cc @nithindavid as it is a UX issue.

**Comment by pranavrajs** [comment]
@danielzzz Thanks for the detailed comment. This is really helpful. I get that the UX around the email channel is confusing.

To answer your questions. You need to set both IMAP and SMTP for an email channel to work. Chatwoot does not ship with a default email server. The forwarder option in the email should not be shown ideally if you are using SMTP/IMAP.

**Comment by danielzzz** [comment]
thanks, but it's strange because I think I tested it with just imap and I believe it was sending messages out.
I will try again, anyway.

**Comment by TroySK** [comment]
I ran into a similar issue yesterday and I kept getting timeout issues in the logs. I have some RoR experience so dig into the code and found things were setup fine. Although as mentioned above the UX/content around the email forwarder and IMAP may be improved.

`Mail.connection {}` kept failing.

The cause of my issue had nothing to do with the app, but the setup. I am using a Dokku instance on a ubuntu machine and had blocked most ports `993` being one of them. Ran `sudo ufw allow 993` and everything started working. As all the replies are in the inbox and unread it successfully imported all our conversations also.
